### PR TITLE
ENB fixes in recv() and recvfrom() handling

### PR DESCRIPTION
**Pull request by aleksander0m**
If recv() or recvfrom() return -1, there should be no chance of treating that return value as a valid amount of data read, or we may be casting it as an unsigned integer and try to process a PDU of 0xFFFFFFFF bytes.

### CODE REVIEW

File: srsenb/src/stack/upper/gtpu.cc [hunk 1]
```diff
@@ -229,10 +229,14 @@ void gtpu::run_thread()
       n = recvfrom(fd, pdu->msg, buflen, 0, (struct sockaddr *)&client, &client_len);
     } while (n == -1 && errno == EAGAIN);
 
-    if (n < 0) {
-        gtpu_log->error("Failed to read from socket\n");
+    if(n < 0) {
```
Comment (andrepuschmann) [inline srsenb/src/stack/upper/gtpu.cc]: please check formating here. Use the .clang-format file in the root directory

File: srsenb/src/stack/upper/gtpu.cc [hunk 2]
```diff
@@ -409,6 +413,15 @@ void gtpu::mch_thread::run_thread()
     do{
       n =  recvfrom(m1u_sd, pdu->msg, SRSENB_MAX_BUFFER_SIZE_BYTES - SRSENB_BUFFER_HEADER_OFFSET, 0, (struct sockaddr *) &src_addr, &addrlen);
     } while (n == -1 && errno == EAGAIN);
+
+    if(n < 0) {
```
Comment (andrepuschmann) [inline srsenb/src/stack/upper/gtpu.cc]: also here format

File: srsenb/src/stack/upper/gtpu.cc [hunk 3]
```diff
@@ -409,6 +413,15 @@ void gtpu::mch_thread::run_thread()
...
+      gtpu_log->error("Failed to read message from socket\n");
+      break;
+    }
+
+    if(n == 0)
```
Comment (andrepuschmann) [inline srsenb/src/stack/upper/gtpu.cc]: we use `if (cond) {}` even for single lines

**Comment by andrepuschmann** [comment]
Hey Aleks, sorry but I just realized that we've also changed the S1AP handling in the new release significantly. I'll close this PR. Can you please check if the issue still exits in the new version and reopen again, if so? Thanks a ton

**Comment by aleksander0m** [comment]
@andrepuschmann is the new version out already?

**Comment by andrepuschmann** [comment]
Not officially, but feel free to have a look at the `merge_master_next` branch. The code you'd be interested in is already there.

**Comment by aleksander0m** [comment]
Yes, I see the code has changed a lot, the fixes are no longer applicable.

**Comment by andrepuschmann** [comment]
Yes, this was my feeling as well. Feel free to test and open the issue again if you still have issues.

**Pull request closed without merging**